internal/sdkprovider/kafkatopicrepository/repository.go

```diff
@@ -0,0 +1,163 @@
+package kafkatopicrepository
+
+import (
+	"context"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/aiven/aiven-go-client/v2"
+)
+
+var (
+	initOnce sync.Once
+	// singleRep a singleton for repository
+	singleRep = &repository{}
+	// errNotFound mimics Aiven "not found" error. Never wrap it, so it can be determined by aiven.IsNotFound
+	errNotFound = aiven.Error{Status: http.StatusNotFound, Message: "Topic not found"}
```

Comment:
consider defining the error constants using `errors.New` or `fmt.Errorf` for more idiomatic Go error creation

Reply:
Please read the description.